packages/pluggableWidgets/popup-menu-web/src/PopupMenu.tsx

```diff
@@ -0,0 +1,141 @@
+                {props.menuTrigger}
+            </div>
+            <div className={`popupmenu-menu popupmenu-position-${props.position}`}>{menuOptions}</div>
+        </div>
+    );
+}
+
+function createMenuOptions(
+    props: PopupMenuContainerProps,
+    handleOnClickItem: (itemAction?: ActionValue) => void
+): ReactElement[] {
+    let menuOptions: ReactElement[];
+
+    if (props.renderMode === "basic") {
+        menuOptions = props.basicItems.map((item, index) => {
+            const pickedStyle =
+                item.styleClass !== "defaultStyle"
+                    ? "popupmenu-basic-item-" + item.styleClass.replace("Style", "")
+                    : "";
+            return item.itemType === "divider" ? (
+                <div key={index} className={"popupmenu-basic-divider"} />
+            ) : (
+                <div
+                    key={index}
+                    className={`popupmenu-basic-item ${pickedStyle}`}
+                    onClick={() => {
+                        handleOnClickItem(item.action);
+                    }}
+                >
+                    {item.caption}
+                </div>
+            );
+        });
+    } else {
+        menuOptions = props.customItems.map((item, index) => (
+            <div key={index} className={"popupmenu-custom-item"} onClick={() => handleOnClickItem(item.action)}>
+                {item.content}
+            </div>
+        ));
+    }
+
+    return menuOptions;
+}
+
+function useOnClickOutside(ref: RefObject<HTMLDivElement>, handler: () => void): void {
+    useEffect(() => {
+        const listener = (event: MouseEvent & { target: Node | null }): void => {
+            if (!ref.current || ref.current.contains(event.target)) {
+                return;
+            }
+            handler();
+        };
+        document.addEventListener("mousedown", listener);
+        document.addEventListener("touchstart", listener);
+        return () => {
+            document.removeEventListener("mousedown", listener);
+            document.removeEventListener("touchstart", listener);
+        };
+    }, [ref, handler]);
+}
+
+function correctPosition(element: HTMLElement): void {
+    const isOffScreen = isElementPartiallyOffScreen(element);
+    console.log(isOffScreen);
+    if (isOffScreen) {
+        moveAbsoluteElementOnScreen(element);
+    }
+    const boundingRect: DOMRect = element.getBoundingClientRect();
+    const blockingElement = isBehindRandomElement(element, boundingRect, 3, "popupmenu");
+    console.log(blockingElement);
+    if (blockingElement && isElementVisibleByUser(blockingElement)) {
+        unBlockAbsoluteElement(element, boundingRect, blockingElement.getBoundingClientRect());
+    } else if (blockingElement) {
+        let node = blockingElement;
+        do {
+            if (isBehindElement(element, node, 1) && isElementVisibleByUser(node)) {
+                unBlockAbsoluteElement(element, boundingRect, node.getBoundingClientRect());
```

Comment:
Maybe return this function invocation iso exlicitly returning `undefined` on the next line?